integrationtest/src/test/resources/immutablesBuilderTest/mapper/src/main/java/org/mapstruct/itest/immutables/style/FixedTopLevelMapper.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.itest.immutables.style;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.ReportingPolicy;
+import org.mapstruct.factory.Mappers;
+import org.mapstruct.itest.immutables.TopLevelDto;
+
+@Mapper(unmappedTargetPolicy = ReportingPolicy.ERROR)
+public interface FixedTopLevelMapper {
+    FixedTopLevelMapper INSTANCE = Mappers.getMapper( FixedTopLevelMapper.class );
+
+    FixedTopLevelWithStyle toImmutable(TopLevelDto dto);
```

Comment:
The reason why it doesn't work for ECJ is this. I don't know why but the `FixedTopLevelWithStyle` is always reported as missing. If you change this to `BaseTopLevelWithStyle` (which is the base class for the generated `FixedTopLevelWithStyle` then it works OK.

Reply:
There's still some underlying root cause here. 

I've changed the usage here as you suggested and the full build works. However, I added a new test case in this PR which works under javac but fails under ECJ to illustrate the problem with a better use case.